src/main/java/br/com/mercadolivre/desafiospring/services/PurchaseOperation.java

```diff
@@ -0,0 +1,76 @@
+package br.com.mercadolivre.desafiospring.services;
+
+import br.com.mercadolivre.desafiospring.models.Product;
+import br.com.mercadolivre.desafiospring.models.Purchase;
+import br.com.mercadolivre.desafiospring.models.PurchaseRequest;
+import br.com.mercadolivre.desafiospring.repository.ApplicationRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PurchaseOperation {
+
+    final private ApplicationRepository<Product, Long> repo;
+
+    public List<Purchase> makePurchase(List<PurchaseRequest> purchaseRequests) {
+
+        List<Purchase> newPurchase = new ArrayList<Purchase>();
+        List<Product> findedProducts = new ArrayList<Product>();
+
+        for (PurchaseRequest pRequests : purchaseRequests
+        ) {
+
+            //TODO: Make a new error handler
+            try {
+                findedProducts = findProducts(pRequests.getProducts());
+
```

Comment:
```suggestion
        List<Purchase> newPurchase = new ArrayList<Purchase>();

        for (PurchaseRequest pRequests : purchaseRequests
        ) {

            //TODO: Make a new error handler
            try {
                List<Product> findedProducts = findProducts(pRequests.getProducts());

```
Visto que está usando o findedProducts dentro do escopo, acredito que seria melhor trazer o newPurchase.add para dentro do Try e criar o findedProducts diretamente no try.

Reply:
Boa!